Libraries/RoslynTripleSlash/TripleSlashSyntaxRewriter.cs

```diff
@@ -100,8 +101,8 @@ internal class TripleSlashSyntaxRewriter : CSharpSyntaxRewriter
 
         private static readonly string[] MarkdownHeaders = new[] { "[!NOTE]", "[!IMPORTANT]", "[!TIP]" };
 
-        // Note that we need to support generics that use the ` literal as well as the escaped %60
-        private static readonly string ValidRegexChars = @"[A-Za-z0-9\-\._~:\/#\[\]\{\}@!\$&'\(\)\*\+,;]|(%60|`)\d+";
+        // Note that we need to support generics that use the ` literal as well as any url encoded character
+        private static readonly string ValidRegexChars = @"[A-Za-z0-9\-\._~:\/#\[\]\{\}@!\$&'\(\)\*\+,;]|`\d+|%\w{2}";
         private static readonly string ValidExtraChars = @"\?=";
 
         private static readonly string RegexDocIdPattern = @"(?<prefix>[A-Za-z]{1}:)?(?<docId>(" + ValidRegexChars + @")+)(?<overload>%2[aA])?(?<extraVars>\?(" + ValidRegexChars + @")+=(" + ValidRegexChars + @")+)?";
```

Comment:
Would the inclusion of `%\w{2}` in the `<docId>` group still allow the `%2[aA]` match to occur for the `<overload>` group? I had shied away from this in #71 because I was worried it would affect the overload handling logic below.

Reply:
Yes it would. I've added logic that handles `*` docId's in the main mapping method. Currently it simply trims the character but based on my feedback in https://github.com/dotnet/runtime/pull/51140 we'd probably to revisit this choice.